### PR TITLE
Maskformer post-processing fixes and improvements

### PR DESCRIPTION
**Pull request by alaradirik**
# What does this PR do?
- Improves MaskFormer docs, corrects minor typos
- Restructures `MaskFormerFeatureExtractor.post_process_panoptic_segmentation` for better readability, adds target_sizes argument for optional resizing
- Adds `post_process_semantic_segmentation` and `post_process_instance_segmentation` methods.
- Adds a deprecation warning to `post_process_segmentation` method in favour of `post_process_instance_segmentation` 

Notes:
This PR is part of a larger effort to ensure consistency of post-processing methods across segmentation models, to define common arguments and outputs, and get ImageSegmentationPipeline working with all available models.
- `post_process_semantic_segmentation` returns segmentations as tensors of shape (height, width), which is consistent with the COCO format
- `post_process_instance_segmentation` returns segmentations either in the same format as the panoptic method or optionally in run-length encoded format (if return_coco_format is set to True).
- `post_process_semantic_segmentation` currently has an inconsistent input argument (target_size instead of target_sizes) and output (3D tensor instead of list of 2D tensors)

## Before submitting
- [X ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [X ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [X ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

### CODE REVIEW

File: src/transformers/models/maskformer/feature_extraction_maskformer.py
Comment (sgugger) [inline src/transformers/models/maskformer/feature_extraction_maskformer.py:517]: Why the new line here? It should be before the comment if we want one.

File: src/transformers/models/maskformer/feature_extraction_maskformer.py
Comment (sgugger) [inline src/transformers/models/maskformer/feature_extraction_maskformer.py:580]:
```suggestion
            target_size (`Tuple[int, int]`, *optional*):
```

File: src/transformers/models/maskformer/feature_extraction_maskformer.py
Comment (sgugger) [inline src/transformers/models/maskformer/feature_extraction_maskformer.py:583]: As mentioned several times before, we need the type before the colon.

File: src/transformers/models/maskformer/feature_extraction_maskformer.py
Comment (sgugger) [inline src/transformers/models/maskformer/feature_extraction_maskformer.py:492]:
```suggestion
            "`post_process_segmentation` is deprecated and will be removed in v5 of Transformers, please use `post_process_instance_segmentation`"
```

File: src/transformers/models/maskformer/feature_extraction_maskformer.py
Comment (sgugger) [inline src/transformers/models/maskformer/feature_extraction_maskformer.py:493]:
```suggestion
            " instead.",
            FutureWarning
```

**Review by sgugger — COMMENTED** [review]
Thanks for the PR! Left a couple of comments.

File: src/transformers/models/maskformer/modeling_maskformer.py
Comment (NielsRogge) [inline src/transformers/models/maskformer/modeling_maskformer.py:271]:
```suggestion
            A tensor of shape `(batch_size, num_queries, num_labels + 1)` representing the class logits for each
            query. Note the `+ 1` is needed because we incorporate the null class.
```

**Review by amyeroberts — COMMENTED** [review]
Thanks for adding these improvements ⭐ 

1. Main comments are about adding tests and structure.
2. Docstring examples for MaskFormer will also need to be updated
3. As mentioned in internal discussions (noting here for visibility), I think there's still some consideration to be had about the naming of the outputs (`segments` vs e.g. `segment_info`) as well as the flag `return_coco_format` as it's the seg maps in coco format, rather than the output itself.

File: src/transformers/models/maskformer/feature_extraction_maskformer.py
Comment (NielsRogge) [inline src/transformers/models/maskformer/feature_extraction_maskformer.py:545]: I don't think `post_process_segmentation` is a public method actually. It is not present in the docs, nor was it shown in any notebook.

So I think we can directly remove it?

Comment (alaradirik) [reply src/transformers/models/maskformer/feature_extraction_maskformer.py:545]: @NielsRogge @sgugger It is used within ImageSegmentationPipeline but I'm simultaneously working on a PR to fix the pipeline. I'll open the other PR once this PR is merged.

I will go ahead and remove `post_process_segmentation` though.

Comment (sgugger) [reply src/transformers/models/maskformer/feature_extraction_maskformer.py:545]: SGTM!

File: src/transformers/models/maskformer/feature_extraction_maskformer.py
Comment (alaradirik) [reply src/transformers/models/maskformer/feature_extraction_maskformer.py]: @amyeroberts makes sense, I restructured this a bit, could you take another look?

Comment (amyeroberts) [reply src/transformers/models/maskformer/feature_extraction_maskformer.py]: Looks good!

**Comment by alaradirik** [comment]
@sgugger @amyeroberts @NielsRogge all comments are addressed, could you approve the PR if everything looks good?

File: src/transformers/models/maskformer/feature_extraction_maskformer.py
Comment (alaradirik) [reply src/transformers/models/maskformer/feature_extraction_maskformer.py:639]: `post_process_segmentation` was included in the model docs so I couldn't remove it.

**Review by sgugger — APPROVED** [review]

**Pull request merged**
